pkg/validation/validation.go

```diff
@@ -14,31 +14,38 @@ import (
 		)
 
 func ValidateSpendTransaction(storage db.Storage, tx *chain.Transaction) (error) {
+        // 300 ns
 	if tx.Body.Output0.Amount.Cmp(big.NewInt(0)) == -1 {
 		return NewErrNegativeOutput(0)
 	}
 	if tx.Body.Output1.Amount.Cmp(big.NewInt(0)) == -1 {
 		return NewErrNegativeOutput(1)
 	}
 
+        // 20 Us
 	prevTx0Conf, err := storage.FindTransactionByBlockNumTxIdx(tx.Body.Input0.BlockNumber, tx.Body.Input0.TransactionIndex)
 	if err == leveldb.ErrNotFound {
 		return NewErrTxNotFound(0, tx.Body.Input0.BlockNumber, tx.Body.Input0.TransactionIndex)
 	}
 	if err != nil {
 		return err
 	}
+
+        // 10 Us
 	prevTx0 := prevTx0Conf.Transaction
 	if prevTx0Conf.ConfirmSigs[0] != tx.Body.Input0ConfirmSig {
 		return NewErrConfirmSigMismatch(0)
 	}
 	sigHash0 := tx.Body.SignatureHash()
+
+        // 200 Us
 	prevTx0Output := prevTx0.Body.OutputAt(tx.Body.Input0.OutputIndex)
 	err = eth.ValidateSignature(sigHash0, tx.Sigs[0][:], prevTx0Output.Owner)
 	if err != nil {
 		return NewErrInvalidSignature(0)
 	}
 
+        // 450 ns
 	totalInput := big.NewInt(0)
 	totalInput = totalInput.Add(totalInput, prevTx0Output.Amount)
 
```

Comment:
Another thing: `if !tx.Body.Input1.IsZero()` never gets tripped which will essentially could double the time because `eth.valdiateSignature` gets called twice.

Reply:
What do you mean never gets tripped? You mean if `Input1` is defined?

Reply:
Yeah if `input1` is never defined. which is the case in our benchmark setup